src/preprocessing/point_in_poly/hierarchical_winding.jl

```diff
@@ -0,0 +1,291 @@
+# This bounding box is used for the hierarchical evaluation of the `WindingNumberJacobsen`.
+# It is implementing a binary tree and thus stores the left and right child and also the
+# faces and closing faces which are inside the bounding box.
+struct BoundingBoxTree{MC, NDIMS}
+    faces         :: Vector{Int}
+    min_corner    :: MC
+    max_corner    :: MC
+    is_leaf       :: Bool
+    closing_faces :: Vector{NTuple{NDIMS, Int}}
```

Comment:
I find it confusing that `faces` is a collection of face indices of the `mesh`, while `closing_faces` is a collection of tuples of vertex indices of the `mesh`.
Can you make both collections of tuples, or would that hurt performance?